### PR TITLE
[FEATURE] Enable self-initializing `ExpectColumnMedianToBeBetween`

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Update Expectation to enable self-initializing capabilities.
- Integration test to ensure that Expectation works when invoked from a Validator.

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py [hunk 1]
```diff
@@ -2176,3 +2176,59 @@ def test_quentin_expect_column_proportion_of_unique_values_to_be_between_auto_ye
 
         max_value_actual: int = result.expectation_config["kwargs"]["max_value"]
         assert max_value_expected >= max_value_actual
+
+
+@pytest.mark.skipif(
+    version.parse(np.version.version) < version.parse("1.21.0"),
+    reason="requires numpy version 1.21.0 or newer",
+)
+@freeze_time(TIMESTAMP)
```
Comment (Shinnnyshinshin) [inline tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py:2201]: ❤️

**Review by Shinnnyshinshin — APPROVED** [review]
This looks great too @cdkini Thank you

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
@cdkini This will need to be updated once #4997 is merged (please use the same order of arguments).  Thank you!

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Pull request merged**